syft/license/license.go

```diff
@@ -0,0 +1,34 @@
+// package license provides common methods for working with SPDX license data
+package license
+
+import (
+	"fmt"
+
+	"github.com/github/go-spdx/v2/spdxexp"
+
+	"github.com/anchore/syft/internal/spdxlicense"
+)
+
+type Type string
+
+const (
+	Declared  Type = "declared"
+	Concluded Type = "concluded"
+)
+
+func ParseExpression(expression string) (string, error) {
+	licenseID, exists := spdxlicense.ID(expression)
+	if exists {
+		return licenseID, nil
+	}
+
+	// If it doesn't exist initially in the SPDX list it might be a more complex expression
+	// ignored variable is any invalid expressions
+	// TODO: contribute to spdxexp to expose deprecated license IDs
```

Comment:
is there an issue in syft or upstream to capture this? could be nice as a reference here (otherwise it will probably get lost)